config/opal_check_libnl.m4

```diff
@@ -0,0 +1,223 @@
+                             [AC_MSG_WARN([Could not link a simple program with lib $1])])
+            AC_LANG_POP(C)
+            ;;
+        esac
+])
+
+dnl
+dnl Check for libnl-3.
+dnl
+dnl Inputs:
+dnl
+dnl $1: prefix where to look for libnl-3
+dnl $2: var name prefix of _CPPFLAGS and _LDFLAGS and _LIBS
+dnl
+dnl Outputs:
+dnl
+dnl - Set $2_CPPFLAGS necessary to compile with libnl-3
+dnl - Set $2_LDFLAGS necessary to link with libnl-3
+dnl - Set $2_LIBS necessary to link with libnl-3
+dnl - Set HAVE_LIBNL3 1 if libnl-3 will be used
+dnl
+AC_DEFUN([OPAL_CHECK_LIBNL3],[
+    OPAL_VAR_SCOPE_PUSH([CPPFLAGS_save LIBS_save LDFLAGS_save])
+    AC_MSG_NOTICE([checking for libnl3])
+
+    AC_MSG_CHECKING([for libnl3 prefix])
+    AC_MSG_RESULT([$1])
+    AC_MSG_CHECKING([for $1/include/libnl3])
+    AS_IF([test -d "$1/include/libnl3"],
+          [opal_libnl3_happy=1
+           AC_MSG_RESULT([found])],
+          [opal_libnl3_happy=0
+           AC_MSG_RESULT([not found])])
+
+    # Random note: netlink/version.h is only in libnl3 - it is not in libnl.
+    # Also, nl_recvmsgs_report is only in libnl3.
+    CPPFLAGS_save=$CPPFLAGS
+    opal_tmp_CPPFLAGS="-I$1/include/libnl3"
+    CPPFLAGS="$opal_tmp_CPPFLAGS $CPPFLAGS"
+    AS_IF([test $opal_libnl3_happy -eq 1],
+          [OPAL_CHECK_PACKAGE([$2],
+                              [netlink/version.h],
+                              [nl-3],
+                              [nl_recvmsgs_report],
+                              [],
+                              [$1],
+                              [],
+                              [opal_libnl3_happy=1],
+                              [opal_libnl3_happy=0])
+
+           # Note that FI_CHECK_PACKAGE is going to add
+           # -I$dir/include into $2_CPPFLAGS.  But because libnl3
+           # puts the headers in $dir/libnl3, we need to
+           # overwrite $2_CPPFLAGS with -I$dir/libnl3.  We can do
+           # this unconditionally; we don't have to check for
+           # success (checking for success occurs below).
+           $2_CPPFLAGS=$opal_tmp_CPPFLAGS])
+
+    # If we found libnl-3, we *also* need libnl-route-3
+    LIBS_save=$LIBS
+    LDFLAGS_save=$LDFLAGS
+    AS_IF([test "$$2_LDFLAGS" != ""],
+          [LDFLAGS="$$2_LDFLAGS $LDFLAGS"])
+    AS_IF([test $opal_libnl3_happy -eq 1],
+          [AC_SEARCH_LIBS([nl_rtgen_request],
+                          [nl-route-3],
+                          [opal_libnl3_happy=1],
+                          [opal_libnl3_happy=0])])
+    LIBS=$LIBS_save
+    LDFLAGS=$LDFLAGS_save
+
+    # Just because libnl* is evil, double check that the
+    # netlink/version.h we found was for libnl3.  As far as we
+    # know, netlink/version.h only first appeared in version
+    # 3... but let's really be sure.
+    AS_IF([test $opal_libnl3_happy -eq 1],
+          [AC_MSG_CHECKING([to ensure these really are libnl3 headers])
+           CPPFLAGS="$$2_CPPFLAGS $CPPFLAGS"
```

Comment:
Minor: here, you unconditionally assign `CPPFLAGS`, but above (lines 150-151), you conditionally assign `LDFLAGS` with the same type of contents.  I don't really care which way you go, but these assignment styles should likely be consistent for ease of future maintenance, IMHO.